src/Controls/samples/Controls.Sample/Startup.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using Maui.Controls.Sample.Pages;
+using Maui.Controls.Sample.Services;
+using Maui.Controls.Sample.ViewModel;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Maui;
+using Microsoft.Maui.Controls.Compatibility;
+using Microsoft.Maui.Hosting;
+
+namespace Maui.Controls.Sample
+{
+	public class Startup : IStartup
+	{
+		public readonly static bool UseXamlPage = false;
+
+		public void Configure(IAppHostBuilder appBuilder)
+		{
+			appBuilder
+				.RegisterCompatibilityRenderers()
+				.UseMauiApp<MyApp>()
```

Comment:
Instead of passing the app into the native `MauiApplication` class, we can allow the user to set it here and it use conditional logic or something.